src/API/Discount.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LemonSqueezy\API;
+
+use function array_map;
+
+use LemonSqueezy\Entity\Discount as DiscountEntity;
+
+class Discount extends AbstractApi
+{
+    public function getAllDiscounts(): array
+    {
+        $discounts = $this->get('/discounts');
```

Comment:
This code would be better:
```suggestion
        $discounts = $this->get('/discounts?include=variants');
```
In the `relationship` part [of the response](https://docs.lemonsqueezy.com/api#including-related-resources) then  a `variant` array appears, which indicates the product variants this discount is limited for.
But: I wasn't sure how to include that in the entity, so I skipped it for now.

Reply:
Let me try to include this now I'm working on it.

Reply:
Ok, let me explain what I think about this. I found will be nice to generate a Fluent API for the whole package that allows you as a developer to do things like this:

```
$lemonSqueeze = $client->fluent();
$variants = ['variants'];
$discountsWithVariants = $client->discounts->withVariant($variants);
```

I think this will resolve how to work with variants and improve the development experience.